### ISSUE TITLE
 a problem pulling the image from dockerhub

### ISSUE THREAD

**Issue by jiatianluo** [umautobots/LONER#13]
I tried to run the command ./build.sh to pull the image from dockerhub, but it didn't finish. I think it might be a network problem, so I changed to the Chinese mirror (because I'm from China). But it still stalled halfway every time. Can you give me some suggestions?

**Comment by sethgi** [comment]
can you give more details about what in particular is failing? Please provide system information, the command you're running, what the error message is, etc.

**Comment by jiatianluo** [comment]
The system does not have any error warnings, but every time I run the ./build.sh command to pull the image, it always stalls halfway through, waiting for a long time without any response, like this
![屏幕截图 2024-01-11 140630](https://github.com/umautobots/LONER/assets/97267545/b0429f14-5fd5-4d43-a7c1-fdd2139d6135)

**Comment by 5hloke** [comment]
Have you modified the container.Dockerfile or the build.sh file in any way ? Additionally it does take some time to build some of the dependencies, so it might just be that you need to wait a little.

**Comment by jiatianluo** [comment]
I haven't made any changes to the Dockerfile or build.sh , and I think it's most likely due to my unstable network. I would like to know if there are any other paths to pull the image

**Comment by sethgi** [comment]
If you're having trouble pulling the base image you can build it from scratch locally. It's not clear to me why you'd be having trouble pulling stuff, since ideally dockerhub should handle that. 

It also remains a bit tricky to help you debug without knowing things like hardware, cuda version, operating system, etc (all of which are fields in our bug report issue template).

The steps are to first run `./build_dockerhub.sh` which builds the image we pushed to dockerhub.

There's a slight problem in that build process where the second to last command (the one that pip installs tiny-cuda-nn) usually fails depending on the system configuration. So what you can do is comment out that RUN command ([this line](https://github.com/umautobots/LONER/blob/e252293b04153f4aed92d4ec8fbd3b4615cc12f8/docker/container_dockerhub.Dockerfile#L64)), let the rest of the build finish, then run the pip command manually and commit the image. It's because of this dependency issue that we ended up pushing the image to dockerhub.

In short:

1. Comment out [this line](https://github.com/umautobots/LONER/blob/e252293b04153f4aed92d4ec8fbd3b4615cc12f8/docker/container_dockerhub.Dockerfile#L64)
2. In a terminal:
```
cd docker
./build_dockerhub.sh
docker run -it --name loner_base_install --gpus=all --runtime=nvidia --privileged loner:base_1.0
sudo ldconfig
pip3 install git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch
<exit container>
docker commit loner_base_install loner:base_1.0
./build.sh
./run.sh
```

Heads up this process is a bit slow, which is why we pushed stuff to dockerhub. But it should work. 

let me know if you have problems... I tested this on my machine but not a clean install or anything so it's possible I missed something.

**Comment by sethgi** [comment]
@jiatianluo Did you manage to sort this out? I'm going to close this issues in a couple days if I don't hear back.
